readthedocs/projects/forms.py

```diff
@@ -394,7 +394,9 @@ def build_versions_form(project):
                 version.identifier[:8],
             )
         else:
-            label = version.verbose_name
+            label = version.slug
+            if version.slug not in version.identifier:
+                label += ' ({})'.format(version.identifier)
```

Comment:
This will show something like `my-slug (a1b2c3d4)`?

Reply:
This only occurs where a version is *not* a tag so generally no. Looking a few lines above might clear this up.

Reply:
It will look like `my-slug (origin/branchname)` typically.